qt-core/src/translation.ts

```diff
@@ -0,0 +1,119 @@
+  findQtPathsInKitDir,
+  IsMacOS,
+  IsWindows,
+  OSExeSuffix,
+  searchForExeInQtInfo,
+  telemetry
+} from 'qt-lib';
+import { coreAPI, projectManager } from '@/extension';
+import path from 'path';
+
+const logger = createLogger('translation');
+
+export async function openInLinguistCommand() {
+  telemetry.sendAction('openInLinguist');
+  const activeDocument = vscode.window.activeTextEditor?.document.uri;
+  if (!activeDocument) {
+    logger.error('No active editor found');
+    return;
+  }
+  const project = projectManager.findProjectContainingFile(activeDocument);
+  if (!project) {
+    logger.error('No project found for the active document');
+    return;
+  }
+  const selectedKitPath = coreAPI?.getValue<string>(
+    project.folder,
+    'selectedKitPath'
+  );
+  const selectedQtPaths = coreAPI?.getValue<string>(
+    project.folder,
+    'selectedQtPaths'
+  );
+  let linguistPath: string | undefined;
+  if (selectedKitPath) {
+    linguistPath = await locateLinguist(selectedKitPath);
+  } else if (selectedQtPaths) {
+    linguistPath = await locateLinguistFromQtPaths(selectedQtPaths);
+  }
+  if (!linguistPath) {
+    logger.error('Linguist executable not found');
+    void vscode.window.showErrorMessage(
+      'Linguist executable not found. Select a Qt Kit with Linguist installed.'
+    );
+    return;
+  }
+  if (!(await exists(linguistPath))) {
+    const err = `Linguist executable not found at ${linguistPath}`;
+    logger.error(err);
+    void vscode.window.showErrorMessage(err);
+    return;
+  }
+  openInLinguist(linguistPath, activeDocument.fsPath);
+}
+
+const LinguistExeName = IsMacOS ? 'Linguist' : 'linguist' + OSExeSuffix;
+
+function getLinguistExePath(selectedQtBinPath: string) {
+  const macOSPath = path.join(
+    'Linguist.app',
+    'Contents',
+    'MacOS',
+    LinguistExeName
+  );
+  return IsMacOS
+    ? path.join(selectedQtBinPath, macOSPath)
+    : path.join(selectedQtBinPath, LinguistExeName);
+}
+
+async function locateLinguistFromQtPaths(selectedQtPaths: string) {
+  const qtInfo = coreAPI?.getQtInfoFromPath(selectedQtPaths);
+  if (!qtInfo) {
+    return undefined;
+  }
+  return searchForExeInQtInfo(qtInfo, getLinguistExePath);
+}
+
+async function locateLinguist(selectedKitPath: string) {
+  let linguistExePath = getLinguistExePath(path.join(selectedKitPath, 'bin'));
+  if (await exists(linguistExePath)) {
+    return linguistExePath;
+  }
+  const qtPaths = findQtPathsInKitDir(selectedKitPath);
+  if (qtPaths) {
+    const linguistPath = await locateLinguistFromQtPaths(qtPaths);
+    if (linguistPath) {
+      return linguistPath;
+    }
+  }
+
+  if (!IsWindows) {
+    linguistExePath = '/usr/bin/linguist';
+    if (await exists(linguistExePath)) {
+      return linguistExePath;
+    }
+  }
+
+  return '';
+}
+
+function openInLinguist(linguistPath: string, file: string) {
+  // Execute the Linguist tool with the specified file
+  const child = child_process.spawn(linguistPath, [file], {
```

Comment:
Does this need the "new" version of linguist? Or can old versions already open files via the command line?